webpack/ForemanTasks/Components/TasksTable/TasksTable.js

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from 'react';
+import { Spinner } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { Table } from 'foremanReact/components/common/table';
+import { STATUS } from 'foremanReact/constants';
+import MessageBox from 'foremanReact/components/common/MessageBox';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { getURIQuery } from 'foremanReact/common/helpers';
+import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';
+import createTasksTableSchema from './TasksTableSchema';
+import { updateURlQuery } from './TasksTableHelpers';
+import './TasksTable.scss';
+
+const TasksTable = ({
+  getTableItems,
+  error,
+  status,
+  results,
+  loading,
+  query,
+  itemCount,
+  pagination,
+}) => {
+  useEffect(() => {
+    getTableItems();
+  }, [query.state, query.result, query.mode, query.time]);
+  const onPageChange = paginationArgs => {
+    updateURlQuery({
+      page: paginationArgs.page,
+      per_page: paginationArgs.perPage,
+    });
+    getTableItems();
+  };
+
+  if (status === STATUS.ERROR) {
+    return (
+      <MessageBox
+        key="tasks-table-error"
+        icontype="error-circle-o"
+        msg={__(`Could not receive data: ${error && error.message}`)}
+      />
+    );
+  }
+
+  if (loading) {
+    return <Spinner size="lg" loading />;
+  }
+
+  if (results.length === 0) {
+    return <span>{__('No Tasks')}</span>;
+  }
+
+  const uriQuery = getURIQuery(window.location.href);
+  return (
+    <React.Fragment>
+      <Table
+        key="tasks-table"
+        columns={createTasksTableSchema(
+          getTableItems,
```

Comment:
Instead of passing the `getTableItems` action, what about passing an action that will just update the redux state with a new `query`. Then, your `useEffect` will call the `getTableItems`?

Reply:
Why? getTableItems updates the state with the new query and results, and useeffect will have to be changed to listen on these changes

Reply:
Added setQuery action